_posts/2023-10-14-bypass-oidc-trust.md

```diff
@@ -0,0 +1,129 @@
+
+So our new trust policy has no access to the `ref` provided by GitHub which now means we have no way to check which branch the request is coming from.
+
+# Testing the process
+
+I've created this repo as an example of how this works.
+
+Firstly we have full branch protection for `main` enabled and we get the approriate error when attempting to push
+
+![Push Denied](/img/bypass-oidc-trust/push-denied.png){:lightbox="true"}
+
+Our AWS Trust policy is configured with branch protection:
+
+![Trust Policy](/img/bypass-oidc-trust/trust-policy.png){:lightbox="true"}
+
+And our Merge is blocked:
+
+![Merge Blocked](/img/bypass-oidc-trust/merge-blocked.png){:lightbox="true"}
+
+The process works wholely as we expect, once the review is complete and the merge happens, the pipeline runes and we get deployed code:
+
+![Pipeline Success](/img/bypass-oidc-trust/pipeline-success.png){:lightbox="true"}
+
+## Malicious user enters the chat
+
+Let's say we see this open source repository, and we want to be a bad person, We can attempt to run our own actions inside a branch, and uses GitHub Actions `on: push` we can essentially run anything we want even in branches
+
+So we create a new branch, make some small modifications to our pipeline config and attempt to deploy code:
+
+![Malicious Changes](/img/bypass-oidc-trust/malicious_changes.png){:lightbox="true"}
+![Nefarious Action Running](/img/bypass-oidc-trust/nefarious_action.png){:lightbox="true"}
+
+We have been thwarted by our OIDC trust policy!
+
+![Thwarted](/img/bypass-oidc-trust/thwarted.png){:lightbox="true"}
+
+## Move to Github Environments
+
+Now, as a Developer, we want to try the Environments feature of Github to manage our deployments and and so we can have different secrets and reuse our pipeline configurations. We enable it, Realise that our Trust Policy no longer works (see above, the change in `sub`) so we update our Trust policy as describe in the [GitHub documentation](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#adding-the-identity-provider-to-aws)
+
+![Updated Trust Policy](/img/bypass-oidc-trust/updated-trust-policy.png){:lightbox="true"}
+
+And add the environment to our pipeline, proceed through the review and merge process, and watch our updated pipeline perform as expected
+
+![Added environment](/img/bypass-oidc-trust/added-environemnt.png){:lightbox="true"}
+![Environment Deployed](/img/bypass-oidc-trust/environment-deploy.png){:lightbox="true"}
+
+## Malicious user number two
+
+Another bad actor notices this new change to use environments, so he attempts to do a deployment aswell.
+
+Remember that now, we are no longer relying on the Trust Policy to limit our credentials to a specific branch because There is no reference to the branch in the Trust Policy due to the limited subset of options provided to us.
```

Comment:
```suggestion
Remember that now, we are no longer able to rely on the Trust Policy to limit our credentials to a specific branch because there is no reference to the branch in the Trust Policy due to the limited subset of options provided to us.
```